web3/utils/compat/compat_codecs.py

```diff
@@ -0,0 +1,24 @@
+
+import codecs
+import sys
+
+
+def backslashreplace(ex):
+    # The error handler receives the UnicodeDecodeError, which contains arguments of the
+    # string and start/end indexes of the bad portion.
+    bstr, start, end = ex.args[1:4]
+
+    # The return value is a tuple of Unicode string and the index to continue conversion.
+    # Note: iterating byte strings returns int on 3.x but str on 2.x
+    return (
+        u''.join(
+            '\\x{:02x}'.format(c if isinstance(c, int) else ord(c))
+            for c in bstr[start:end]
+        ),
+        end,
+    )
+
+
+def setup_backslashreplace():
+    if sys.version_info[:2] < (3, 5):
+        codecs.register_error('backslashreplace', backslashreplace)
```

Comment:
Does anything bad happen if this happens to be called twice?

Reply:
Probably just a performance penalty. I would guess that the error handling in the default implementation is pretty optimized.